rinja/src/filters/builtin.rs

```diff
@@ -10,6 +10,28 @@ use crate::{Error, Result};
 // MAX_LEN is maximum allowed length for filters.
 const MAX_LEN: usize = 10_000;
 
+/// Return an ephemeral `&str` for `$src: impl fmt::Display`
+///
+/// If `$str` is `&str` or `String`, this macro simply passes on its content.
+/// If it is neither, then the formatted data is collection into `&buffer`.
+///
+/// `return`s with an error if the formatting failed.
+macro_rules! try_to_str {
+    ($src:expr => $buffer:ident) => {
+        match format_args!("{}", $src) {
+            args => {
+                if let Some(s) = args.as_str() {
```

Comment:
I didn't know about this method on `Arguments`. Nice!

Reply:
Yeah, it's a case of "wouldn't it be nice if there was a function that ...". And then the function exists and it does just work. :)